controllers/Weathers.js

```diff
@@ -0,0 +1,44 @@
+const BaseController = require('./Base');
+const View = require('../views/Base');
+const weatherUtils = require('../models/weather-utils.js');
+
+const ZONE_CODE_LENGTH = 10;
+const REQ_DATE_LENGTH = 8;
+
+module.exports = BaseController.extend({
+	name: "Weathers",
+	run: function(req, res, next) {
+		const zoneCode = req.params.zoneCode;
+		const reqDate = req.params.date
+		if (!isValidZoneCode(zoneCode) || !isValidReqDate(reqDate)) {
+			res.send("Invalid zone code or date");
+			return;
+		}
+		const date = convertDateToDbForm(reqDate);
+		weatherUtils.loadWeather(date, zoneCode).then(function(err, weather) {
+			const refinedWeather = weatherUtils.convertWeatherToClientWeatherAPIResponse(weather);
+			res.jsonp(refinedWeather);
+		}).catch(function(err) {
+			console.log("Db error");
```

Comment:
You should respond to client's request even when there's an error.
See [here](https://github.com/Ques-Study/WeatherCalendarServer/wiki/Client-api-structure#failure) for more detail.